toolkit/tools/imagegen/configuration/parse_partition.go

```diff
@@ -87,7 +87,28 @@ func processDisk(inputDiskValue string) (err error) {
 			disks = append(disks, Disk{})
 		}
 
-		disks[latestDiskIndex].PartitionTableType = PartitionTableTypeGpt
+		// In kickstart installation scenario, the partition table type is set to
+		// MBR by default. The Anaconda installer has this config "--gpt" that indicates
+		// whether the users prefer creation of GPT disk label or not. The value of "--gpt"
+		// is a bool where "True" indicates using GPT and "False" if not, which means using MBR.
+		// This config is set as a boot option within /proc/cmdline, which will be parsed by anaconda
+		// during installation process. Thus, Mariner will also pick the same design to reach compatibility
+		// with kickstart scenario
+
+		// Please note that this code is only executed during kickstart installation, when "IsKickStartBoot" is set to true.
+		// Mariner installer currently does not allow direct specification of disk and partition layout within
+		// the image config file for kickstart installation. So any disk/partition setting you make in the image config file
+		// will be overwritten if you enable kickstart installation mode.
```

Comment:
Useful comment, thank you!

Considering what you've written, let's also add (assuming it's not there already) logic to the config validation, where having both `IsKickStartBoot` and disk/partition layout is considered invalid and immediately errors out.

Reply:
Have already added logic to capture this case in imageconfigvalidator. Here's the link for reference: https://github.com/microsoft/CBL-Mariner/blob/7b9a7b0992e9d98b5f37bbfbc11289502d1b2375/toolkit/tools/imageconfigvalidator/imageconfigvalidator.go#L82